File: buildSrc/src/main/groovy/org/mockito/release/comparison/PublicationsComparator.java
package org.mockito.release.comparison;

import groovy.lang.Closure;

import java.io.File;

/**
 * Compares publications
 */
public interface PublicationsComparator {

    /**
     * Sets files for comparison
     */
    void compareBinaries(Closure<File> left, Closure<File> right);

    /**
     * Sets poms for comparison
     */
    void comparePoms(Closure<String> left, Closure<String> right);

    /**
     * Gives information if publications are equal
     */
    boolean isPublicationsEqual();
}

File: buildSrc/src/main/kotlin/org/mockito/release/comparison/PublicationsComparator.kt
package org.mockito.release.comparison

import groovy.lang.Closure

import java.io.File

/**
 * Compares publications
 */
interface PublicationsComparator {

    /**
     * Sets files for comparison
     */
    fun compareBinaries(left: Closure<File>, right: Closure<File>)

    /**
     * Sets poms for comparison
     */
    fun comparePoms(left: Closure<String>, right: Closure<String>)

    /**
     * Gives information if publications are equal
     */
    var publicationsEqual: Boolean?
}
